tooling/debugger/src/context.rs

```diff
@@ -55,25 +64,36 @@ impl<'a, B: BlackBoxFunctionSolver> DebugContext<'a, B> {
         }
     }
 
+    pub(super) fn get_current_source_location(&self) -> Option<Vec<Location>> {
+        self.get_current_opcode_location()
+            .as_ref()
+            .and_then(|location| self.debug_artifact.debug_symbols[0].opcode_location(location))
+    }
```

Comment:
I think we should add some documentation here around why this is an option. i.e. that some opcodes are not tied to a particular line of code due to constraining circuit inputs, etc.